test/check/check.go

```diff
@@ -75,6 +75,61 @@ func CheckComponentsRemoval(components []Component) {
 	}
 }
 
+func getConfigComponentsMap(gvk schema.GroupVersionKind) (map[string]struct{}, error) {
+	config := GetConfig(gvk)
+	if config == nil {
+		return nil, fmt.Errorf("config not found")
+	}
+	configV1 := ConvertToConfigV1(config)
+	existingComponentsMap := map[string]struct{}{}
+
+	existingComponentsMap[MonitoringComponent.ComponentName] = struct{}{}
+
+	if configV1.Spec.KubeMacPool != nil {
+		existingComponentsMap[KubeMacPoolComponent.ComponentName] = struct{}{}
+	}
+	if configV1.Spec.KubevirtIpamController != nil {
+		existingComponentsMap[KubevirtIpamController.ComponentName] = struct{}{}
+	}
+	if configV1.Spec.KubeSecondaryDNS != nil {
+		existingComponentsMap[KubeSecondaryDNSComponent.ComponentName] = struct{}{}
+	}
+	if configV1.Spec.LinuxBridge != nil {
+		existingComponentsMap[LinuxBridgeComponent.ComponentName] = struct{}{}
+	}
+	if configV1.Spec.MacvtapCni != nil {
+		existingComponentsMap[MacvtapComponent.ComponentName] = struct{}{}
+	}
+	if configV1.Spec.Ovs != nil {
+		existingComponentsMap[OvsComponent.ComponentName] = struct{}{}
+	}
+	if configV1.Spec.Multus != nil {
+		existingComponentsMap[MultusComponent.ComponentName] = struct{}{}
+	}
+	if configV1.Spec.MultusDynamicNetworks != nil {
+		existingComponentsMap[MultusDynamicNetworks.ComponentName] = struct{}{}
+	}
+	if configV1.Spec.MultusDynamicNetworks != nil {
+		existingComponentsMap[MultusDynamicNetworks.ComponentName] = struct{}{}
+	}
```

Comment:
dup

Reply:
fixed! thanks

Reply:
interesting if there is a tool that can detect copy paste dups like this
sometimes it makes bugs, here it was just cosmetics

Reply:
perhaps introducing golint will be a good idea for CNAO, but it's not on the near future roadmap as it will take time to refactor the code.

Reply:
ye no rush about it imho as well